File: AuctionFE/src/app/models/category.ts
import { Auction } from './auction';
export class Category {
    id: number;
    name: string;
    description: string;
    auction: Auction[];
    
    constructor(id: number,
        name: string,
        description: string,
        auction: Auction[]) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.auction = auction;
    }
}

File: src/main/java/com/auctionwebsite/model/Category.java
package com.auctionwebsite.model;

import lombok.*;

import javax.persistence.*;
import java.io.Serializable;

@Data
@NoArgsConstructor
@Builder
@EqualsAndHashCode(of = {"id"})
@ToString(exclude = {"auction"})
@AllArgsConstructor
@Entity
@Table(name = "category")
public class Category implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;
    @Column(name = "name")
    private String name;
    @Column(name = "description")
    private String description;
    @OneToOne( cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JoinColumn(name = "id", referencedColumnName = "category_id")
    private Auction auction;
}
